tests/Backend/Bigquery/TableDefinitionOperationsPartitioningTest.php

```diff
@@ -0,0 +1,218 @@
+                    'partitions' => [
+                        // todo: expected one partition https://keboola.atlassian.net/browse/BIG-186
+                        // same apply for partitions after snapshot restore
+                    ],
+                ]
+            ),
+            $tableResponse['definition']
+        );
+
+        // test snapshots
+        $snapshotId = $this->_client->createTableSnapshot($tableId);
+        $newTableId = $this->_client->createTableFromSnapshot($tableResponse['bucket']['id'], $snapshotId, 'restored');
+        $newTable = $this->_client->getTable($newTableId);
+        $this->assertSame($expectedTableDefinition, $newTable['definition']);
+
+        $workspace = $this->initTestWorkspace();
+        // test workspace load
+        // if we would implement copy load this needs to be also tested
+//        $workspaces = new Workspaces($this->_client);
+//        $workspaces->loadWorkspaceData($workspace['id'], [
+//            'input' => [
+//                [
+//                    'source' => $tableId,
+//                    'destination' => 'partitioned',
+//                ],
+//            ],
+//        ]);
+
+        // test unload table from WS
+        $backend = WorkspaceBackendFactory::createWorkspaceBackend($workspace);
+        assert($backend instanceof BigqueryWorkspaceBackend);
+
+        $client = $backend->getDb();
+
+        $dataset = $client->dataset($workspace['connection']['schema']);
+        $tableInWorkspace = $dataset->table('partitioned');
+        if ($tableInWorkspace->exists()) {
+            $tableInWorkspace->delete();
+        }
+        $dataset->createTable('partitioned', [
+            'schema' => [
+                'fields' => [
+                    [
+                        'name' => 'id',
+                        'type' => 'INTEGER',
+                        'mode' => 'REQUIRED',
+                    ],
+                    [
+                        'name' => 'time',
+                        'type' => 'TIMESTAMP',
+                        'mode' => 'REQUIRED',
+                    ],
+                ],
+            ],
+            'timePartitioning' => [
+                'type' => 'DAY',
+                'field' => 'time',
+                'expirationMs' => 1000,
+            ],
+            'clustering' => [
+                'fields' => ['id'],
+            ],
+        ]);
+
+        // unload table back to storage
+        $unloadedTableId = $this->_client->createTableAsyncDirect($this->getTestBucketId(self::STAGE_IN), [
+            'name' => 'partitionedUnload',
+            'dataWorkspaceId' => $workspace['id'],
+            'dataTableName' => 'partitioned',
+        ]);
+        $unloadedTable = $this->_client->getTable($unloadedTableId);
+        // table created by unload from WS are created without type
+        $this->assertArrayNotHasKey('definition', $unloadedTable);
+    }
+
+    public function testErrorWhenCreatingTableWithPartitioning(): void
```

Comment:
Hmm tento test mi lokalne pada
```
Failed asserting that 'Failed to create table "my_new_table" in dataset "in_c_API_tests_3c0d1c044f50a86492e311f213364346cf460963". Exception: Either interval partition or range partition should be specified.. Requested table: {\n
    "schema": {\n
        "fields": [\n
            {\n
' matches PCRE pattern "/Failed to create table "my_new_table" in dataset ".*"\. Exception: The field specified for clustering cannot be found in the schema\..*/".

```

Reply:
Už asi vím :) já mám ve vendoru novů utils libku co vyžaduje php8.1 :D pustím si to znova a opravím

Reply:
Tak ne mě to frčí a ten test prochází

Reply:
škoda že nešel na testingu :cry: 
2023-10-11T09:10:35.5577562Z 08:33:15-UTC - paratest-bigquery                  -	2) Keboola\Test\Backend\Bigquery\TableDefinitionOperationsPartitioningTest::testErrorWhenCreatingTableWithPartitioning
2023-10-11T09:10:35.5582696Z 08:33:15-UTC - paratest-bigquery                  -	Keboola\StorageApi\ClientException: You don't have access to the resource.

Reply:
daj si ještě composer install v connection podle mě máš starý driver

Reply:
Jj to som prave pozeral ci to na testingu preslo :/

Reply:
Jo super preslo to dik.